src/iminuit/cost.py

```diff
@@ -106,7 +106,38 @@ def _soft_l1_loss(z_sqr):
 def _soft_l1_cost(y, ye, ym):
     return _soft_l1_loss(_z_squared(y, ye, ym))
 
+def bblite(mu,n):
+    """
+    Compared to the "normal" Beeston-Barlow method, the lite method uses one nuisance parameter (NP) per bin instead of one NP per component per bin.
+    Beeston-Barlow assumed a Poisson distribution.
+    Signal and background templates share one parameter per bin.
+    For a single bin, i:
+                -ln(L) = -n*ln(b*mu) + b*mu + (b-1)^2/2s^2 ,   (1)
+    [note that we emit the label i in all variables (L, b,u,s and n) for simplicity]
+    where L is the likelihood,
+    n is the number of yields,
+    b is the nuisance parameter (NP) (only 1 used)-->what we calculate,
+    mu is the total number of expected events in the bin, given the values of all the other parameters,
+    and s (sigma) is the relative (statistical) uncertainty in the prediction.
+
+
+    To minimise this log-likelihood, we set the derivative of (1) to zero, i.e. -ln(L)/dB = 0
+
+                    ==> b^2 + (mu*s^2 - 1)b - n*s^2 = 0.        (2)
+
+    One can then solve the quadratic equation and obtain the correct (maximal/positive)root.
+        
+        ==> b = -(mu*s^2-1)/2  +/- sqrt( (mu*s^2-1)^2 - 4n*s^2 ) /2
+                = -    x /2    +/-  sqrt ( x^2       - 4n*s^2 ) /2    (3)
+                
+                ,where x = (mu*s^2-1).
+    """
+    s2 = 1/mu
+    x = mu * s2 - 1
+    b = -x * 0.5 + np.sqrt(x**2 + 4 * n * s2) * 0.5
```

Comment:
Ok, I think I understand this. x is zero for poisson errors, which you assume here, but not in general. I started to implement the more general case. No need to edit your code.

Reply:
Having thought more about this, I think this is not correct, because `mu` is a mixture of different templates, each with a different statistical error. You need to perform error propagation to get the uncertainty of `mu` based on the uncertainties of the templates and the component fractions.

Reply:
Hi @HDembinski. Apologies for the delayed response. I checked out this out and had a discussion with my supervisor. We are assuming a Poisson error. However, we were supposed to use the relative uncertainty, instead we just used the normal standard deviation (1/sqrt(N)), which is incorrect. Additionally, as you said, `mu` is a mixture of templates so one needs to determine the corresponding uncertainties and evaluate the value for `s`, where `s = uncertainties/templates`, so `s2 = (uncertainties/templates)^2 = variances/templates^2`